backend/cmm_helpers.ml

```diff
@@ -750,34 +753,41 @@ let mod_int c1 c2 is_safe dbg =
    occur, in which case we force x / -1 = -x and x mod -1 = 0. (PR#5513). *)
 
 let is_different_from x = function
-  | Cconst_int (n, _) -> n <> x
-  | Cconst_natint (n, _) -> n <> Nativeint.of_int x
+  | Cconst_int (n, _) -> Nativeint.of_int n <> x
+  | Cconst_natint (n, _) -> n <> x
```

Comment:
why is this change here ? does this actually fix a bug ? if so, that deserves a comment